src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/QueueHealthDecider.java

```diff
@@ -93,13 +99,16 @@ private void configureActionPriority() {
    */
   private Action computeBestAction(NodeKey esNode, ResourceEnum threadPool) {
     Action action = null;
-
-    for (String actionName : actionsByUserPriority) {
-      action =
-        getAction(actionName, esNode, threadPool, true);
-      if (action != null) {
-        break;
+    if (canUseMoreHeap(esNode)) {
+      for (String actionName : actionsByUserPriority) {
+        action = getAction(actionName, esNode, threadPool, true);
+        if (action != null) {
+          break;
+        }
       }
+    } else {
+      PerformanceAnalyzerApp.RCA_RUNTIME_METRICS_AGGREGATOR.updateStat(
+          RcaRuntimeMetrics.NO_INCREASE_ACTION_SUGGESTED, NAME + ":" + esNode.getHostAddress(), 1);
```

Comment:
Should we differentiate between an increase in the QueueIncreaseAction and Cache Increase Action?

Reply:
Right, the `NAME ` part does that.